modules/waf/main.tf

```diff
@@ -2,8 +2,56 @@
 # To enable/disable Web Application Firewall(WAF) for a domain
 ##############################################################################
 
+locals {
+  rulesets_list = data.ibm_cis_rulesets.rulesets.rulesets_list
+  rulesets_map  = { for rule in local.rulesets_list : rule.name => rule.ruleset_id }
+
+  # Filter and construct the rules array
+  rules = [
+    for rule in var.enabled_rulesets : contains(keys(local.rulesets_map), rule) ? {
+      id         = local.rulesets_map[rule]
+      enabled    = true
+      expression = "true"
+    } : null
+  ]
+}
+
 resource "ibm_cis_domain_settings" "domain_settings" {
+
+  count     = !var.disable_legacy_waf ? 1 : 0
   cis_id    = var.cis_instance_id
   domain_id = var.domain_id
   waf       = var.enable_waf ? "on" : "off"
 }
+
+moved {
+  from = ibm_cis_domain_settings.domain_settings
+  to   = ibm_cis_domain_settings.domain_settings[0]
+}
+
+data "ibm_cis_rulesets" "rulesets" {
+  cis_id    = var.cis_instance_id
+  domain_id = var.domain_id
+}
+
+resource "ibm_cis_ruleset_entrypoint_version" "config" {
+  cis_id    = var.cis_instance_id
+  domain_id = var.domain_id
+  phase     = "http_request_firewall_managed"
+
+  rulesets {
+    description = "Entry Point ruleset"
```

Comment:
I think that we dont need to hardcode the description. Is this description per ruleset? or one for all?